changelog/8014.bugfix.md

```diff
@@ -0,0 +1,16 @@
+Setting `model_confidence=cosine` in `DIETClassifier`, `ResponseSelector` and `TEDPolicy` is deprecated and will be removed in the next Rasa Open Source minor version `2.4.0`. Experiments suggest that using cosine similarity as model's confidences can change the ranking of predicted labels. Additionally, using `model_confidence=inner` for `TEDPolicy` is discouraged as it currently breaks the action selection logic in the ensemble of policies.
```

Comment:
and can we say when we introduced the cosine thing and phrase it more as that we made a mistake?